sw/device/lib/aes.c

```diff
@@ -26,7 +26,8 @@ void aes_init(aes_cfg_t aes_cfg) {
   REG32(AES_CTRL_SHADOWED(0)) = cfg_val;
 };
 
-void aes_key_put(const void *key, aes_key_len_t key_len) {
+void aes_key_put(const void *key_share0, const void *key_share1,
```

Comment:
We might need/want another function that takes a key that's not in shares, pulls in entropy/mask and puts mask in share0, key_input ^ mask in share1?

Sometimes we are given a key in raw form and need a way to feed it in.

Reply:
I think we really don't want to give the impression that this is something to do lightly. But I also see that this might be needed. Maybe this is something to keep in mind when updating the DIF? I am thus also pinging @silvestrst . But he is on vacation right now, so we might need to re-discuss this later.

Reply:
I think the DIF will always present a 2-share interface, and layers on top would have to manage the 1-share variant (because that requires interacting with other hardware, something DIFs are not allowed to do).

Reply:
Ah good point about the interaction with the other hardware.  Your proposal for the layers on top to handle the 1-share is sensible.